### PR TITLE
kubeadm: add 1.18 entry in the etcd map

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add 1.18 entry in the etcd map

**Which issue(s) this PR fixes**:
xref #https://github.com/kubernetes/kubeadm/issues/1807

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @neolit123
/priority important-soon
/area kubeadm

### CODE REVIEW

**Review by ereslibre — APPROVED** [review]
Thank you @fabriziopandini!

/lgtm
/approve
/hold

Feel free to unhold.

File: cmd/kubeadm/app/constants/constants.go
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go]: @fabriziopandini  we can remove `12: "3.2.24",` once this merges:
https://github.com/kubernetes/test-infra/pull/14483

Comment (fabriziopandini) [reply cmd/kubeadm/app/constants/constants.go]: ok let's wait few days then

Comment (neolit123) [reply cmd/kubeadm/app/constants/constants.go]: that PR just merged

**Comment by fabriziopandini** [comment]
@neolit123 removed the entry for 1.12 as well

**Comment by fabriziopandini** [comment]
@neolit123 @ereslibre 
make tests pass and refactor upgrade test so they will automatically stay up to grade with new releases

**Comment by neolit123** [comment]
/lgtm

> make tests pass and refactor upgrade test so they will automatically stay up to grade with new releases

awesome
xref https://github.com/kubernetes/kubeadm/issues/1260

**Review by rosti — COMMENTED** [review]
Thanks @fabriziopandini !
I like the refactoring, but can we split the commits in a couple of different PRs?

**Comment by fabriziopandini** [comment]
@rosti 
The etcd map change and test refactoring are already in separated PR.
Test refactoring is the biggest one, but TBH, being a refactor of a single func I find a little bit forceful to split it in separated PR...

**Comment by rosti** [comment]
Ok, let's merge it like this then. We did a few reviews of this already and it looks good.
/hold cancel

**Pull request merged**
